From 430c7309b245d55762729bb0ac5927d24e42699e Mon Sep 17 00:00:00 2001
From: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Date: Mon, 29 Jul 2019 12:41:38 +0530
Subject: [PATCH] Fix for crash in Car settings after Network Reset

Issue:  Bluetooth paired devices are not removed from
paired list after network reset in Car Settings

Reason: This is due to NullPointerException which occurs
during telephony reset.

Fix: Check for Telephony feature support for IVI devices

Change-Id: If1cb94c26043855fad16a6e3e16f9c5d5718b4e7
Tracked-On:
Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
---
 .../system/ResetNetworkConfirmFragment.java        | 30 ++++++++++++----------
 1 file changed, 17 insertions(+), 13 deletions(-)

diff --git a/src/com/android/car/settings/system/ResetNetworkConfirmFragment.java b/src/com/android/car/settings/system/ResetNetworkConfirmFragment.java
index 11d2240..ef94d4e 100644
--- a/src/com/android/car/settings/system/ResetNetworkConfirmFragment.java
+++ b/src/com/android/car/settings/system/ResetNetworkConfirmFragment.java
@@ -21,6 +21,7 @@ import android.bluetooth.BluetoothAdapter;
 import android.bluetooth.BluetoothManager;
 import android.content.ContentResolver;
 import android.content.Context;
+import android.content.pm.PackageManager;
 import android.content.SharedPreferences;
 import android.net.ConnectivityManager;
 import android.net.NetworkPolicyManager;
@@ -83,6 +84,7 @@ public class ResetNetworkConfirmFragment extends SettingsFragment {
         }
 
         Context context = requireActivity().getApplicationContext();
+        PackageManager pm = context.getPackageManager();
 
         ConnectivityManager connectivityManager = (ConnectivityManager)
                 context.getSystemService(Context.CONNECTIVITY_SERVICE);
@@ -105,21 +107,23 @@ public class ResetNetworkConfirmFragment extends SettingsFragment {
             }
         }
 
-        int networkSubscriptionId = getNetworkSubscriptionId();
-        TelephonyManager telephonyManager = (TelephonyManager)
-                context.getSystemService(Context.TELEPHONY_SERVICE);
-        if (telephonyManager != null) {
-            telephonyManager.factoryReset(networkSubscriptionId);
-        }
+        if (pm.hasSystemFeature(PackageManager.FEATURE_TELEPHONY)) {
+            int networkSubscriptionId = getNetworkSubscriptionId();
+            TelephonyManager telephonyManager = (TelephonyManager)
+                    context.getSystemService(Context.TELEPHONY_SERVICE);
+            if (telephonyManager != null) {
+                telephonyManager.factoryReset(networkSubscriptionId);
+            }
 
-        NetworkPolicyManager policyManager = (NetworkPolicyManager)
-                context.getSystemService(Context.NETWORK_POLICY_SERVICE);
-        if (policyManager != null) {
-            String subscriberId = telephonyManager.getSubscriberId(networkSubscriptionId);
-            policyManager.factoryReset(subscriberId);
-        }
+            NetworkPolicyManager policyManager = (NetworkPolicyManager)
+                    context.getSystemService(Context.NETWORK_POLICY_SERVICE);
+            if (policyManager != null) {
+                String subscriberId = telephonyManager.getSubscriberId(networkSubscriptionId);
+                policyManager.factoryReset(subscriberId);
+            }
 
-        restoreDefaultApn(context, networkSubscriptionId);
+            restoreDefaultApn(context, networkSubscriptionId);
+        }
 
         // There has been issues when Sms raw table somehow stores orphan
         // fragments. They lead to garbled message when new fragments come
-- 
2.7.4

